recipes/nsync/all/conanfile.py

```diff
@@ -106,3 +106,5 @@ def _def_compoment(self, name, lib=None):
 
         if self.settings.os in ["Linux", "FreeBSD"]:
             component.system_libs = ["pthread"]
+            if name == "nsync_cpp":
```

Comment:
honestly, all this package_info() could be summarized in few lines without this `_def_compoment()` function:

```python
    def package_info(self):
        self.cpp_info.components["nsync_c"].libs = ["nsync"]
        self.cpp_info.components["nsync_cpp"].libs = ["nsync_cpp"]
        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.components["nsync_c"].system_libs.append("pthread")
            self.cpp_info.components["nsync_cpp"].system_libs.extend(["m", "pthread"])
```

all the `cmake_find_package` stuff is uncessary since they is no config file upstream.

Reply:
@SpaceIm, I'm still very new with conan so I'd rather stick with the existing code for this part than break something downstream.
If you can push to my branch (not sure the "allow edit by maintainers" apply to you), then feel free to do so.
Otherwise, I'll wait for another review to either leave as-is or take your proposed code as-is.

Reply:
@SpaceIm, I reworked by flattening then removing the extra stuff. If other reviews disagree I can still revert the commit removing the extra stuff but I'm going with you initial assessment.